google/cloud/pubsub_v1/subscriber/_protocol/streaming_pull_manager.py

```diff
@@ -435,48 +552,78 @@ def _schedule_message_on_hold(
         assert self._callback is not None
         self._scheduler.schedule(self._callback, msg)
 
-    def _send_unary_request(self, request: gapic_types.StreamingPullRequest) -> None:
+    def send_unary_ack(
+        self, ack_ids, future_reqs_dict
+    ) -> Tuple[List[requests.AckRequest], List[requests.AckRequest]]:
         """Send a request using a separate unary request instead of over the stream.
 
-        Args:
-            request: The stream request to be mapped into unary requests.
+        If a RetryError occurs, the manager shutdown is triggered, and the
+        error is re-raised.
         """
-        if request.ack_ids:
-            self._client.acknowledge(  # type: ignore
-                subscription=self._subscription, ack_ids=list(request.ack_ids)
+        assert ack_ids
+
+        error_status = None
+        ack_errors_dict = None
+        try:
+            self._client.acknowledge(subscription=self._subscription, ack_ids=ack_ids)
+        except exceptions.GoogleAPICallError as exc:
+            _LOGGER.debug(
+                "Exception while sending unary RPC. This is typically "
+                "non-fatal as stream requests are best-effort.",
+                exc_info=True,
             )
+            error_status = _get_status(exc)
+            ack_errors_dict = _get_ack_errors(exc)
+        except exceptions.RetryError as exc:
+            _LOGGER.debug(
```

Comment:
The comment concerns be: what is shutting down?

Reply:
It's the bidi stream itself. If no progress whatsoever can be made - no calls are succeeding - we shut down the stream. The user receives an error message for the entire subscribe process. ie: there's a subscribe future that users can use to track the entire subscribe process / stream itself, and that is failed. So users will have to recreate the subscriber client. We should leave this behavior as is.

I believe I'm missing the completion of the futures with a failure in this case.

I think we should leave this behavior as is.